src/adr/0010-enable-elasticsearch-routing.md

```diff
@@ -0,0 +1,41 @@
+# 10 Enable ElasticSearch routing
+
+Date: 2019-10-17
+
+## Status
+
+Proposed
+
+Additional performance testing is required for adoption.
+
+## Context
+
+Our queries are mostly bounded to a mailbox or a user. We can easily
+limit the number of ElasticSearch nodes involved in a given query by
+grouping the underlying documents on the same node using a routing key.
+
+Without routing key, each shard needs to execute the query. The coordinator
+needs also to be waiting for the slowest shard.
+
+Using the routing key unlocks significant throughput enhancement (proportional
```

Comment:
I understand the latency thing but I don't get why the routing would enhance scalability

Reply:
The low level example I provided is not relevant in an ADR.

Please refer to the pull request comment explaining that: https://github.com/linagora/james-project/pull/2757#issuecomment-545366619